src/main/java/com/github/shiranr/scooters/db/CosmosClient.java

```diff
@@ -0,0 +1,55 @@
+package com.github.shiranr.scooters.db;
+
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import com.github.shiranr.scooters.domain.internal.Scooter;
+
+import java.util.Optional;
+
+import static java.net.HttpURLConnection.HTTP_ACCEPTED;
+import static java.net.HttpURLConnection.HTTP_CREATED;
+import static java.net.HttpURLConnection.HTTP_OK;
+
+public class CosmosClient implements Client<Scooter>{
+
+    private Connection connection;
+
+    public CosmosClient(Connection connection) {
+        this.connection = connection;
+    }
+
+    public Scooter[] getAll() {
+        String query = "SELECT * FROM " + connection.getContainerName();
+        CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
+        CosmosPagedIterable iterator = connection.getContainer().queryItems(query, options, Scooter.class);
+        return (Scooter[]) iterator.stream().toArray(Scooter[]::new);
+    }
+
+    public Scooter get(String id) {
+        String query = "SELECT * FROM " + connection.getContainerName() + " c WHERE c.id='" + id + "'";
+        CosmosQueryRequestOptions options = new CosmosQueryRequestOptions().setMaxBufferedItemCount(1);
+        CosmosPagedIterable iterator = connection.getContainer().queryItems(query, options, Scooter.class);
+        Optional first = iterator.stream().findFirst();
+        if (first.isPresent()) {
+            return (Scooter) first.get();
+        }
+        return null;
```

Comment:
not sure about returning null. maybe an error/exception?

Reply:
it's a matter of implemetation IMO - if the requesting process knows how to handle a null result than that's ok (possibly add to the documetation). else need to add expection handling through all the application process.

Reply:
I checked the methods which are calling this function, they are all checking equality to null and if so, they throw and exception.
So for readability I moved the exception into the method.